joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala

```diff
@@ -159,17 +159,15 @@ trait TypeRecoveryParserConfig[R <: X2CpgConfig[R]] { this: R =>
 abstract class XTypeRecovery[CompilationUnitType <: AstNode](cpg: Cpg, state: XTypeRecoveryState) extends CpgPass(cpg) {
 
   override def run(builder: DiffGraphBuilder): Unit = {
-   compilationUnit
-      .map(unit => generateRecoveryForCompilationUnitTask(unit, builder).fork())
-      .map(_.get)
-      .reduceOption((a, b) => a || b)
-      .getOrElse(false)
+    for(cu <- compilationUnits){
```

Comment:
I want to measure the performance impact if I had to fix the multi-threading in a branch from this one, if it goes without a hitch (i.e., tests don't start failing for some reason), then I may be able to have a swift follow-up for a faster overall pass

Reply:
I'll follow this up relatively quickly with a version that is potentially parallel. Could you rubber-stamp https://github.com/ShiftLeftSecurity/codepropertygraph/pull/1748 for that?